message_ix_models/tools/costs/projections.py

```diff
@@ -458,8 +468,21 @@ def create_message_outputs(
         )
         .astype(dtypes)
         .query("year_vtg in @config.Y")
+        .assign(first_technology_year=lambda x: x.first_technology_year.astype(float))
+        .assign(first_technology_year=lambda x: x.first_technology_year.astype(int))
```

Comment:
Is there any specific need for these two chained calls instead of directly?
```
.astype({"first_technology_year": int})
```

If there is, it would be good to record in a comment so people maintaining the code can understand why.